From b2810bf0cb6184501f357c854a8e7408d2dfc84f Mon Sep 17 00:00:00 2001
From: Sagi Maimon <smaimon@habana.ai>
Date: Sun, 31 Jan 2021 15:42:44 +0200
Subject: [PATCH] [SW-31361] gaudi: check svn max limit

For security reasons, when loading fit image,
preboot should check that the fit SVN does not exceed max SVN limit.

Change-Id: Iddf85758de5f508df53c799311a66fd28e2b2355
Signed-off-by: Sagi Maimon <smaimon@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index c75b3e2..1735946 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -238,10 +238,13 @@
 	if (svn < 0)
 		return false;
 
-	if (sec_ver < svn)
-		preboot_puts_hex("Security version too low: ", sec_ver, 0);
+	if (sec_ver < svn || sec_ver > SVN_MAX_VERSIONS) {
+		preboot_puts_hex("Illegal security version: ",
+				sec_ver, false);
+		return false;
+	}
 
-	return sec_ver >= svn;
+	return true;
 }
 
 /* Enable bit is SOC_STRAP10 in schematics */
